ocaml/xapi/xapi_pool_update.ml

```diff
@@ -480,6 +480,22 @@ let resync_host ~__context ~host =
     (* Clean updates that don't have a corresponding patch record *)
     Db.Pool_update.get_all ~__context
     |> List.filter (fun self -> Xapi_pool_patch.pool_patch_of_update ~__context self = Ref.null)
+    |> List.iter (fun self -> destroy ~__context ~self);
+
+    (*
+     * If db indicates an update is not applied to any host but the corresponding patch is applied
+     * on some host(s), that means the RPU has completed, the update record should be removed.
+     * (The patch record will be removed along with the update record.)
+    *)
+    Db.Pool_update.get_all ~__context
+    |> List.filter (fun self ->
```

Comment:
This could be further reduced into one logical formula without if-then-else:

```
  |> List.filter (fun self ->
      Db.Pool_update.get_hosts ~__context ~self = [] 
      && Xapi_pool_patch.pool_patch_of_update ~__context self
         |> fun self -> Db.Pool_patch.get_host_patches ~__context ~self
         |> function [] -> false | _ -> true)

The filter predicate is true if the list of pool-update hosts is empty and the list of host-patches is not empty. Is this the correct logic? I suggest @jonludlam takes a look, too.
```